### PR TITLE
Link cross-sells product image to product page

### LINKED ISSUE #42032: Link cross-sells product images to corresponding product pages
## Is your feature request related to a problem? Please describe.

Currently, the product images of the cross-sells products are not linked to the product page, while the product images within the cart line items section and various other blocks, are linked to the product page. Apart from that, when using the shortcode-based cart, the product images of the cross-sells products are also linked.

## Describe the solution you'd like

I expect that the product images within the cross-sells products on the Cart block are linked to the corresponding product page.

### PR DESCRIPTION
**Pull request by nielslange**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes #42032.

When using the classic cart, the product images of the cross-sells products are linked to the corresponding product pages. When using the Cart block, the product images of the cross-sells products are not linked. This PR synchs the behaviour between the classic cart and the Cart block.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Ensure that one product has at least one cross-sells product, and that cross-sells product is on sale. See https://woo.com/document/related-products-up-sells-and-cross-sells/
2. Go to the frontend and add the edited product to the cart.
3. Go to the Cart block and verify that the product title of the cross-sells product links to the corresponding product page, and the "Sale" badge displays properly.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [x] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [x] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

Link cross-sells product images to corresponding product pages.

#### Comment 

</details>

### CODE REVIEW

**Review by wavvves — APPROVED** [review]
Testing checks out and code changes look clean.
I also tested for the appearance of the appearance of the Sale badge, since that has changes it could be included in the testing instruction so I'll be updating it.

image

I also updated this with trunk so we can make sure CI results are up to date if unrelated, so please wait for them to run before merge 👍🏼

**Comment by wavvves** [comment]
CI passed 🥳

**Comment by nielslange** [comment]
> Testing checks out and code changes look clean. I also tested for the appearance of the appearance of the Sale badge, since that has changes it could be included in the testing instruction so I'll be updating it.
> 
> I also updated this with trunk so we can make sure CI results are up to date if unrelated, so please wait for them to run before merge 👍🏼

Thanks for reviewing this PR, updating the testing instructions and merging the trunk into it. 🙌

**Pull request merged**
